### DISCUSSION TITLE
m2e not resolving project dependency jars

### DISCUSSION THREAD

**Discussion by rotty3000** [eclipse-m2e/m2e-core#717] [Q&A]
It seems that in m2e when a plugin with a lifecycleMappingMetadata is executed, intended for the package phase, the dependency resolver does not resolve project dependencies as built jars, but rather as their outputdirectory.

However, everything works from the maven command line as expected, including when I build the project separated from the reactor build (say using the -pl flag). In that case all project dependencies that can be resolved to jars from the local cache are used if present before any outputdirectory is used. m2e does not do this and seems to default to the outputdirectory which feels like a discrepancy. This has been manifesting in recent releases. I'm wondering if a recent change may have caused this change in behaviour? I'm currently running with the latest: M2E - Maven Integration for Eclipse (includes Incubating components) 1.20.1.20220227-1319 org.eclipse.m2e.feature.feature.group Eclipse.org - m2e

**Comment by laeubi** [comment]
Can you provide a small example to reproduce this? Currently m2e "builds" each project separately and if you enable workspace resolving map dependencies to workspace projects so maybe that is causing what you see with the output directory?

**Comment by rotty3000** [comment]
> and if you enable workspace resolving map dependencies to workspace projects

this I can check. Where is this option? I did not find anything that resembled this description in the m2e settings.

**Comment by rotty3000** [comment]
oh, I think it is this... for each project:
![Screenshot from 2022-05-17 14-10-27](https://user-images.githubusercontent.com/146764/168881320-99748618-ddf4-4aa5-83f2-e0fef4d917bc.png)

Is there a global way to disable that?

**Comment by laeubi** [comment]
You can right click on the Project and choose Maven > Enable Workspace resolution, not sure if this works for multiple projects. There was already a request for a workspace settings here
- https://github.com/eclipse-m2e/m2e-core/issues/547

but it got not much attention yet.
